### PR TITLE
Redirect gRPC server logs to the Datadog logger

### PR DESCRIPTION
**Pull request by juliogreff**
### What does this PR do?

Back in 975fd365ee7e8e84a9bf2736333a83244a7b0f8f we did the same thing
for the gRPC clients, but hadn't noticed the same problem occurred with
the server. To be able to use the same solution across both, now we have
a `pkg/util/grpc` package that also became a little smarter to avoid
duplicating logging lines for higher severities.

This has a small change in behavior as well as it supports
`GRPC_GO_LOG_SEVERITY_LEVEL` and `GRPC_GO_LOG_VERBOSITY_LEVEL` env
variables for finer grained control of gRPC logs.

### Describe your test plan

This change affects both the agent gRPC server, and the clients in the process/trace agent with the remote tagger enabled (#7208), so we need to verify that all behave as expected.

*  By default, no INFO or WARNING gRPC logs should be emitted. You can grep them by "ccResolverWrapper".
* With `GRPC_GO_LOG_SEVERITY_LEVEL` set to "info", you should start seeing gRPC logs. Additionally, they need to be in the correct format as described in #7357.

### CODE REVIEW

File: pkg/util/grpc/logger.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	errorW := ioutil.Discard
+	warningW := ioutil.Discard
+	infoW := ioutil.Discard
+
+	// grpc-go logs to an io.Writer on a certain severity, and every single
+	// lower severity (so FATAL would log to FATAL, ERROR, WARN and INFO),
+	// causing a lot of duplication, and that's why we parse the severity
+	// out of the log instead of having an output for each severity
+
+	logLevel := os.Getenv("GRPC_GO_LOG_SEVERITY_LEVEL")
```
Comment (clamoriniere) [inline pkg/util/grpc/logger.go:66]: maybe a simpler solutions should be to lower the `logLevel` string.

File: pkg/util/grpc/logger.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	switch logLevel {
+	case "", "ERROR", "error": // If env is unset, set level to ERROR.
+		errorW = newRedirectLogger()
+	case "WARNING", "warning":
+		warningW = newRedirectLogger()
+	case "INFO", "info":
+		infoW = newRedirectLogger()
+	}
```
Comment (clamoriniere) [inline pkg/util/grpc/logger.go:74]: nits: maybe a simpler solutions should be to lower the `logLevel` string.

```suggestion
	logLevel := strings.ToLower(os.Getenv("GRPC_GO_LOG_SEVERITY_LEVEL"))
	switch logLevel {
	case "", "error": // If env is unset, set level to ERROR.
		errorW = newRedirectLogger()
	case "warning":
		warningW = newRedirectLogger()
	case "info":
		infoW = newRedirectLogger()
	}
```

**Review by clamoriniere — APPROVED** [review]
small nit comment. otherwise looks good

**Review by sgnn7 — APPROVED** [review]
@juliogreff Awesome work! Ty! 🚀

**Pull request merged**